Ginger/Ginger/WindowExplorer/Common/WindowExplorerPage.xaml.cs

```diff
@@ -839,20 +839,25 @@ private void SetControlsGridView()
             WindowControlsGridView.InitViewItems();
         }
 
-        private void RefreshControlsGrid()
+        private async void RefreshControlsGrid()
         {
-            if (WindowsComboBox.SelectedValue != null && mWindowExplorerDriver != null)
+            if (mWindowExplorerDriver != null)
             {
-                List<ElementInfo> list = mWindowExplorerDriver.GetVisibleControls(CheckedFilteringCreteriaList.Select(x => x.ElementType).ToList());
-                                
+                List<ElementInfo> list = await Task.Run(() => mWindowExplorerDriver.GetVisibleControls(CheckedFilteringCreteriaList.Select(x => x.ElementType).ToList()));
                 // Convert to obserable for the grid
```

Comment:
why the check for "WindowsComboBox.SelectedValue != null " was removed?

Reply:
Because its impossible to interact with a UI element in async function

Reply:
but it was added from some reason so we can't just avoid and remove this check.
so do it by storing the WindowsComboBox.SelectedValue on global field and update it on WindowsComboBox.SelectionChanged event and then check global field is null in IF

Reply:
Corrected